File: backend/backend/src/SignalingComponent/Common/Api/DTOs/ConnectionResponseMessage.cs
using System.Text.Json.Serialization;
using backend.WebSocketComponent.Common.Api.DTOs;

namespace backend.SignalingComponent.Common.Api.DTOs;

public class ConnectionResponseMessage : ITypedMessage
{
    public static string TypeString => "connection-response";

    [JsonIgnore]
    public string InstanceTypeString => TypeString;

    [JsonPropertyName("accepted")]
    public bool Accepted { get; set; }

    [JsonPropertyName("remoteToken")]
    public required string RemoteToken { get; set; }
}

File: frontend/src/types/connection/ConnectionResponseMessage.ts
import { ITypedMessage } from "../ITypedMessage";
import { MessageType } from "../MessageType";

export class ConnectionResponseMessage implements ITypedMessage {
    public readonly type = MessageType.CONNECTION_RESPONSE;
    public msg: {
        accepted: boolean;
        remoteToken: string;
    };

    public constructor(msg: { accepted: boolean; remoteToken: string }) {
        this.msg = msg;
    }
}
